src/api/app/models/package.rb

```diff
@@ -947,7 +947,7 @@ def jobhistory_list(project, repository, arch, options = {})
   end
 
   def service_error(revision = nil)
-    revision ||= serviceinfo['xsrcmd5']
+    revision ||= serviceinfo.try { to_hash['xsrcmd5'] }
```

Comment:
I don't think this is the best solution. I think that the `serviceinfo` method must return a `{}` if there is no `dir['serviceinfo]` is `nil`.

I mean, `serviceinfo` returns a `{}` if there is an exception, so I guess that the default behaviour is that.

Reply:
that was the idea yes - so ``dir['serviceinfo'] || {}``

Reply:
Yo, for sure not! This is

1) for documentation of the monkey patch
2) I would nevertheless propose to merge it to have a deployable master

Reply:
Btw: @mdeniz @saraycp 

This is the output of the Directory.hashed

```
X({"name"=>"kernel-default", "rev"=>"2", "vrev"=>"2", "srcmd5"=>"aae8998dace7a7eb6087d1e362925326", "linkinfo"=>X({"project"=>"home:duwe:branches:Kernel:HEAD", "package"=>"kernel-source", "error"=>"home:duwe:branches:Kernel:HEAD/kernel-source: conflict in file config.tar.bz2"}), "entry"=>X({"name"=>"_link", "md5"=>"581cdadd84890e029027bdc9cde9b707", "size"=>"46", "mtime"=>"1250198147"})})
```

So we should check for error here and return empty hash then?

Reply:
That error is part of the `linkinfo` key. Maybe it is better not to rely on an error inside 'linkinfo` but instead focus on the `serviceinfo` key.

Reply:
ah you're right @saraycp ... nevermind!